frontend/src/stimulus/controllers/pattern-input.controller.ts

```diff
@@ -499,7 +515,7 @@ export default class PatternInputController extends Controller {
             node.after(document.createTextNode(CONTROL_SPACE));
           }
 
-          if (this.isText(follower) && !this.isWhitespaceOrControlSpace(follower.wholeText[0])) {
+          if (this.isText(follower) && !this.isWhitespaceOrControlSpace(follower.wholeText[0] || '')) {
```

Comment:
:question: I am a bit surprised about this change, considering that I am looking at TypeScript.

Apparently `[0]` can in some cases (e.g. empty string) return `undefined`. However, `isWhitespaceOrControlSpace` only accepts arguments of type string. What's surprising to me is that this didn't result in a TS error before, because we are clearly passing `string|undefined` into a method that accepts `string`.

In general I am wondering whether we should solve it as done here (`|| ""`) or whether we should simply accept undefined/null as a parameter as well and explicitly return false in that case. Something similar was done to many of the `is*` methods in this controller.